lib/trino-record-decoder/src/main/java/io/trino/decoder/protobuf/ProtobufColumnDecoder.java

```diff
@@ -69,43 +79,51 @@ public ProtobufColumnDecoder(DecoderColumnHandle columnHandle)
             this.columnType = columnHandle.getType();
             this.columnMapping = columnHandle.getMapping();
             this.columnName = columnHandle.getName();
+            this.oneOfs = columnHandle.getOneOfs()
+                    .map(list -> list.stream().sorted(String::compareTo).collect(toImmutableList()));
```

Comment:
why do we need to sort?

Reply:
I was sorting it based on [this comment in JsonType](https://github.com/trinodb/trino/blob/f854e103364f67e9c90c1aa3387d1bf00664dda8/core/trino-main/src/main/java/io/trino/type/JsonType.java#L37). Maybe it isn't relevant if the built type is a `Slice` instead of a `Block`.

Reply:
this is very interesting, I had no idea the Trino JsonType required keys to be sorted. I now see in JDBC connectors we don't need to sort because it's done by `JsonTypeUtils#jsonParse`/`JsonTypeUtils#toJsonValue` for us.

Reply:
Maybe we should encode this knowledge?

Reply:
Since the JSON record we create will only have at most one field in it (since `oneof` can only have one set), sorting the column handles isn't necessary here.